### DISCUSSION TITLE
Merging is blocked - Missing successful Production deployment

### DISCUSSION THREAD

**Discussion by mirhamasala** [vercel/community#3215] [Help] [answered]
### Problem Description

On Pull Requests, I'm seeing `Merging is blocked - Missing successful Production deployment`

I can see the Preview Deployment and link, though.

Is this expected, and is there any way to remove the block? 

<img width="925" alt="Screenshot 2023-07-03 at 12 12 46" src="https://github.com/vercel/vercel/assets/26620750/98935332-8e2b-4fba-aa95-f3338073b111">

**Comment by amyegan** [comment, marked as answer]
Sorry for the delayed response, @mirhamasala. I just noticed your question posted over in the CLI repo, so I moved it over here to our main discussion board where I spend most of my time. 🙂 

It looks like the branch requires a Production deployment before merging, but the default Vercel Git integration only creates a Preview deployment until the code is actually merged into your production branch. You have a couple of options to solve this.

The first option, and the one I recommend, is to change the branch's settings in GitHub to only require a successful Preview deployment and not a Production deployment. After the PR is merged, the production deployment will happen automatically.

The other option requires a bit more setup [using a GitHub workflow to deploy to production](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel) or manual deployment to Production.

**Reply by mirhamasala** [reply]
Ah, super clear. Thank you. ☀️

I went with the first solution, as you suggested.

If someone else is looking for this, it's on one's repo in GitHub under:

Settings > Branches > Branch Protection Rules > Edit > Require deployments to succeed before merging

<img width="768" alt="Screenshot 2023-07-13 at 17 44 09" src="https://github.com/vercel/community/assets/26620750/f7fd2ae4-3eca-4d58-9fdb-b78d672da358">

**Reply by Kristi47** [reply]
I have the same issue with Vercel and Github actions. 

I have configured Vercel to only build main branch/meaning production env

On github I have added a github action on every pull request. Basically what it does, its on every pull request t .main branch it will run prettier, check commit lint and if those 2 jobs are successful it will pull the env variablea from Vercel Preview env, it will run a build using the preview flag and it will push to vercel preview env.

At the end it will set the generated vercel URL for the preview env as a url for git preview env

Everything was working as expected until it stopped working.

Right now the main branch rulea requires a successful deployment to the preview env, which is actually happening. But still github does not allow branch to be merged. The message says it requires a successfull deployment to preview env.

**Reply by fractionfitness** [reply]
I have the same problem as [Kristi47](https://github.com/Kristi47). I have a long-running dev and main (default) branch.  My repo is setup to require successful preview deployments before merging into dev and main. 

When I open a PR (e.g. feat branch into dev), I use github actions to deploy to vercel because I do a lot of configuration when I deploy feat branches. Even though the vercel deployment to preview was successful, github still blocks the merge because of "Missing successful Preview deployment.".  

I have a feeling this is a problem with the vercel integration. If you do not use Vercel's automated deployment (Vercel Github integration) and, instead, you use vercel cli through github actions, Github is not notified that a preview deployment was made.  However, when I check the deployments for my Github Preview environment, it says that the deployment was successful based after the successful workflow run. My solution for now is to just turn off requiring successful deployments for PRs.

Note: 
My github environments match my vercel environments as follows: 
gh main (production) => vercel (production)
gh dev => vercel (preview on dev branch)
gh feat => vercel (preview on feat branch)

**Reply by fractionfitness** [reply]
[Kristi47](https://github.com/Kristi47), I just solved this by using "Preview" environment in github instead of lowercase "preview". Vercel environments are still lowercase (e.g. preview). 

Github capitalizes the environment name even though I set it at "preview" initially. You will have to delete your existing Github "preview" environment before you can add a new environment called "Preview"
![image](https://github.com/vercel/community/assets/123066159/198ce2f5-95e3-451e-83c6-f4f09806e7ca)

After changing it to "Preview", my status checks passed
![image](https://github.com/vercel/community/assets/123066159/f37c3e33-0648-40da-8ac3-e5aa13459df9)

Note that somehow, Github environment staging is still in lowercase. So the exceptions must be Preview and Production, which need to be capitalized.
